### ISSUE TITLE
Hello Receive examples

### ISSUE THREAD

**Issue by matoxbr** [fphammerle/python-cc1101#41]
Guys in the examples folder just found it for shipping. 
Does anyone have an example to receive?
thanks

**Comment by fphammerle** [comment]
Hi,

receiving is not officially supported yet.
I implemented some basic support (private / unstable API).

You can receive packages by calling `transceiver._wait_for_packet(timeout_seconds=21, gdo0_gpio_line_name="GPIO25")` (or similar)

Example: https://github.com/fphammerle/wireless-sensor/blob/v0.4.0/wireless_sensor/__init__.py#L150

Currently, I cannot guarantee backward-compatibility in future releases yet (thus, the method has a `_` prefix)

**Comment by oragamiman01** [comment]
Hello,

Do you plan to add receiving support? :)

Thanks

**Comment by fphammerle** [comment]
Yes, I do. Actually, I have been using `transceiver._wait_for_packet` for several months already but I haven't found time to make the method / API public yet.

**Comment by Crsarmv7l** [comment]
> Yes, I do. Actually, I have been using `transceiver._wait_for_packet` for several months already but I haven't found time to make the method / API public yet.

I assume you are using the FIFO buffer sure to pythons slow speed?

**Comment by fphammerle** [comment]
Yes, see `CC1101._get_received_packet`. In summary:
`self._read_burst(start_register=FIFORegisterAddress.RX, length=self._read_status_register(StatusRegisterAddress.RXBYTES))`

**Comment by balaboooot** [comment]
Hello! I have a 433mhz WS90 weather station, a Rspi zero2, a cc1101. I tried to receive the signal the way you did but failed. I just need a simple example to receive fsk2 signal. Thanks!

**Comment by balaboooot** [comment]
Hello! I have a WS90 weather station, a Respi zero2w, a cc1101. I tried to receive the signal the way you did but failed. So I just need a simple example to recive fsk2 signal. Thanks!
